packages/core/src/components/DataTable/index.tsx

```diff
@@ -179,22 +224,27 @@ export class DataTable extends React.Component<DataTableProps & WithStylesProps,
     }
   };
 
-  private expandRow = (newExpandedRowIndex: number) => (
+  private expandRow = (newExpandedRowOriginalIndex: number) => (
     event: React.SyntheticEvent<EventTarget>,
   ) => {
     event.stopPropagation();
     this.setState(({ expandedRows }) => {
       const newExpandedRows = new Set(expandedRows);
-      if (expandedRows.has(newExpandedRowIndex)) {
-        newExpandedRows.delete(newExpandedRowIndex);
+      if (expandedRows.has(newExpandedRowOriginalIndex)) {
+        newExpandedRows.delete(newExpandedRowOriginalIndex);
       } else {
-        newExpandedRows.add(newExpandedRowIndex);
+        newExpandedRows.add(newExpandedRowOriginalIndex);
       }
 
       return {
         expandedRows: newExpandedRows,
       };
     });
+
+    setTimeout(() => {
```

Comment:
could also add a comment here for the timeout/race condition, and maybe use `forceUpdate` as miles said above (I forget if that worked or not).

Reply:
@schillerk can you add a comment please?

Reply:
ditto only need to call this if `dynamicRowHeight === true`

Reply:
ditto - wrap around with an `if (dynamicRowHeight)`